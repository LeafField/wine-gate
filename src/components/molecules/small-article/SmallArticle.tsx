import React, { FC } from "react";
import Link from "next/link";
import Pill from "../../atoms/pill/Pill";
import WineImage from "../../atoms/wine-image/WineImage";
import { SmallArticleProps } from "../../../types/article_types";

const SmallArticle: FC<SmallArticleProps> = ({
  tags,
  image_src,
  title,
  id,
  categories,
}) => {
  const splitTags = tags && tags.split(",");

  return (
    <Link
      href={`/article/${id}`}
      className="group row-span-3 grid h-full w-full grid-rows-subgrid gap-y-0 overflow-hidden border border-gray transition-colors duration-500 hover:border-blue"
      aria-label={title}
    >
      <WineImage src={image_src} />
      <h3 className="w-full px-2 text-base text-black">{title}</h3>
      <div className="mt-auto flex flex-row justify-center gap-2 pb-2 pt-2">
        {categories && <Pill title={categories.chara} />}
        {splitTags && splitTags.map((tag) => <Pill key={tag} title={tag} />)}
      </div>
    </Link>
  );
};

export default SmallArticle;
