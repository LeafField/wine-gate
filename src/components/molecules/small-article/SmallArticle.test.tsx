import { render, screen } from "@testing-library/react";
import SmallArticle from "./SmallArticle";

describe("SmallArticle 単体テスト", () => {
  const props = {
    chara: "chara",
    tags: "tag1,tag2",
    image_src: "/test/image.jpg",
    title: "Test Title",
    id: "1",
  };

  test("指定されたプロパティを持つ要素がレンダリングされているか", () => {
    render(<SmallArticle {...props} />);
    const imageElement = screen.getByAltText("ワイン画像");
    expect(imageElement).toBeInTheDocument();

    const titleElement = screen.getByText(props.title);
    expect(titleElement).toBeInTheDocument();

    const charaElement = screen.getByText(props.chara);
    expect(charaElement).toBeInTheDocument();

    const tagElements = screen.getAllByText(/tag/);
    expect(tagElements.length).toBe(2);
  });

  test("スナップショット", () => {
    const { container } = render(<SmallArticle {...props} />);
    expect(container).toMatchSnapshot();
  });
});
