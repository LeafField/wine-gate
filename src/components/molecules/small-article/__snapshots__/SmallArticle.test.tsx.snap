// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallArticle 単体テスト スナップショット 1`] = `
<div>
  <a
    aria-label="Test Title"
    class="group row-span-3 grid h-full w-full grid-rows-subgrid gap-0 border border-gray transition-colors duration-500 hover:border-blue"
    href="/article/1"
  >
    <div
      class="aspect-video w-full overflow-hidden bg-gray"
    >
      <figure
        class="relative h-full w-full transition-transform duration-300 group-hover:scale-110"
      >
        <img
          alt="ワイン画像"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="(min-width:768px) 332px,100vw"
          src="/_next/image?url=%2Ftest%2Fimage.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Ftest%2Fimage.jpg&w=16&q=75 16w, /_next/image?url=%2Ftest%2Fimage.jpg&w=32&q=75 32w, /_next/image?url=%2Ftest%2Fimage.jpg&w=48&q=75 48w, /_next/image?url=%2Ftest%2Fimage.jpg&w=64&q=75 64w, /_next/image?url=%2Ftest%2Fimage.jpg&w=96&q=75 96w, /_next/image?url=%2Ftest%2Fimage.jpg&w=128&q=75 128w, /_next/image?url=%2Ftest%2Fimage.jpg&w=256&q=75 256w, /_next/image?url=%2Ftest%2Fimage.jpg&w=384&q=75 384w, /_next/image?url=%2Ftest%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=%2Ftest%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=%2Ftest%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=%2Ftest%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ftest%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ftest%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ftest%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ftest%2Fimage.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </figure>
    </div>
    <h3
      class="w-full px-2 text-base text-black"
    >
      Test Title
    </h3>
    <div
      class="mt-auto flex flex-row justify-center gap-2 pb-2 pt-2"
    >
      <div
        class="w-fit rounded-[3.2rem] border border-black px-2 py-[0.125rem] text-black text-sm"
      >
        Test Chara
      </div>
      <div
        class="w-fit rounded-[3.2rem] border border-black px-2 py-[0.125rem] text-black text-sm"
      >
        tag1
      </div>
      <div
        class="w-fit rounded-[3.2rem] border border-black px-2 py-[0.125rem] text-black text-sm"
      >
        tag2
      </div>
    </div>
  </a>
</div>
`;
